api/src/modules/projects/projects.service.ts

```diff
@@ -262,4 +237,69 @@ export class ProjectsService extends AppBaseService<
     );
     this.logger.warn(`Computed and saved ${fromExcel.length} projects`);
   }
+
+  public async createProject(
+    createProjectDto: CreateProjectDto,
+  ): Promise<Project> {
+    const scoreCardRating = await this.scorecard.getRating(createProjectDto);
+    const costs =
+      await this.projectCalculation.computeCostForProject(createProjectDto);
+    const projectSize = await this.dataSource
+      .getRepository(ProjectSize)
+      .findOne({
+        select: ['sizeHa'],
+        where: {
+          ecosystem: createProjectDto.ecosystem,
+          activity: createProjectDto.activity,
+          countryCode: createProjectDto.countryCode,
+        },
+      });
+
+    // TODO: Not clear if sizeHa has to come from the DTO or be retrieved from the database. Does it make sense to have it in the DB?
+    //       Would make sense to have thresholds defined?
+    const project = new ProjectBuilder(
+      createProjectDto,
+      scoreCardRating,
+      costs,
+      projectSize.sizeHa,
+    ).build();
+    return this.projectRepository.save(project);
+  }
+
+  public async updateProject(
+    id: string,
+    updateProjectDto: UpdateProjectDto,
+  ): Promise<Project> {
+    const projectToUpdate = await this.projectRepository.findOneBy({ id });
```

Comment:
Little detail to add to your arsenal (in case u didnt already know), the repository api does provide a .findOneOrFail method which will launch an exception, for the cases that you want to have a custom handling for it (i.e when you don't want to bubble this exception up to the consumer)